Shared/Flows/Onboarding/OnboardingCoordinator.swift

```diff
@@ -21,40 +21,142 @@ class OnboardingCoordinator: PresentableCoordinator<Void> {
     }
 
     override func start() {
+        self.setInitialOnboardingContent()
         self.handle(deeplink: self.deepLink)
     }
 
     func handle(deeplink: DeepLinkable?) {
         guard let link = deeplink else { return }
+
         self.onboardingVC.reservationId = link.reservationId
         self.onboardingVC.passId = link.passId
         self.onboardingVC.updateUI()
     }
-}
 
-extension OnboardingCoordinator: LaunchActivityHandler {
+    // MARK: - Onboarding Flow Logic
 
-    nonisolated func handle(launchActivity: LaunchActivity) {
-        Task.onMainActor {
-            self.onboardingVC.handle(launchActivity: launchActivity)
+    private func setInitialOnboardingContent() {
+        let userStatus = User.current()?.status
+
+        let initialContent: OnboardingContent
+        switch userStatus {
+        case .needsVerification, .inactive, .waitlist, .none:
+            initialContent = .welcome(self.onboardingVC.welcomeVC)
+        case .active:
+            self.finishFlow(with: ())
+            return
+        }
+
+        self.onboardingVC.switchTo(initialContent)
+    }
+
+    /// Returns the content for the first incompleted onboarding step in the onboarding sequence.
+    private func getNextIncompleteOnboardingContent() -> OnboardingContent? {
+        guard let current = User.current(), let status = current.status else {
+            // If there is no user, then they'll need to provide a phone number to create one.
+            return .phone(self.onboardingVC.phoneVC)
+        }
+
+        switch status {
+        case .needsVerification:
+            return .code(self.onboardingVC.codeVC)
+        case .waitlist:
+            if !current.fullName.isValidPersonName {
+                return .name(self.onboardingVC.nameVC)
+            } else if current.smallImage.isNil {
+                return .photo(self.onboardingVC.photoVC)
+            } else {
+                return .waitlist(self.onboardingVC.waitlistVC)
+            }
+        case .inactive:
+            if !current.fullName.isValidPersonName {
+                return .name(self.onboardingVC.nameVC)
+            } else if current.smallImage.isNil {
+                return .photo(self.onboardingVC.photoVC)
+            } else {
+                return nil
+            }
+        case .active:
+            // Active users don't need to do onboarding.
+            return nil
```

Comment:
All the logic needed to determine the next onboarding step is now in one function.